pkg/readiness/ready_tracker.go

```diff
@@ -333,6 +345,32 @@ func (t *Tracker) collectInvalidExpectations(ctx context.Context) {
 	}
 }
 
+func (t *Tracker) trackAssignMetadata(ctx context.Context) error {
+	defer func() {
+		t.assignMetadata.ExpectationsDone()
+		log.V(1).Info("AssignMetadata expectations populated")
+
+		_ = t.constraintTrackers.Wait()
+	}()
+
+	if !*mutation.MutationEnabled {
+		return nil
+	}
+
+	assignMetadataList := &mutationv1alpha.AssignMetadataList{}
+	lister := retryLister(t.lister, retryAll)
+	if err := lister.List(ctx, assignMetadataList); err != nil {
+		return fmt.Errorf("Error listing AssignMetadata: %w", err)
+	}
+	log.V(1).Info("setting expectations for AssignMetadata", "AssignMetadata Count", len(assignMetadataList.Items))
+
+	for _, am := range assignMetadataList.Items {
+		log.V(1).Info("expecting AssignMetadata", "name", am.GetName())
+		t.assignMetadata.Expect(&am)
```

Comment:
Not quite sure under what circumstances this would change, and even if, I assume the test or a first attempt to use gk would catch that.
Also, changing this for all trackers deserves a new PR.

Reply:
> what circumstances this would change

Code evolution sometimes leads to changes in assumptions.

> Also, changing this for all trackers deserves a new PR.

Agreed.